packages/ui/src/views/user-management/components/dialogs/components/reset-password/reset-password.tsx

```diff
@@ -0,0 +1,76 @@
+import { Button, ButtonGroup, CopyButton, Dialog, Input } from '@/components'
+import { IResetPasswordDialogProps } from '@/views/user-management/components/dialogs/components/reset-password/types'
+import { useStates } from '@/views/user-management/providers/state-provider'
+import { useUserManagementStore } from '@/views/user-management/providers/store-provider'
+
+export function ResetPasswordDialog({ handleUpdatePassword, open, onClose }: IResetPasswordDialogProps) {
+  const { useTranslationStore, useAdminListUsersStore } = useUserManagementStore()
+  const { t } = useTranslationStore()
+  const { user, generatePassword, setGeteneratePassword, password } = useAdminListUsersStore()
+
+  const { loadingStates, errorStates } = useStates()
+  const { isUpdatingUser } = loadingStates
+  const { updateUserError } = errorStates
+
+  const onSubmit = () => {
+    handleUpdatePassword(user?.uid || '')
+    setGeteneratePassword(true)
+  }
+
+  return (
+    <Dialog.Root open={open} onOpenChange={onClose}>
+      <Dialog.Content className="max-w-xl">
+        <Dialog.Header>
+          <Dialog.Title className="font-medium">
+            {t('views:userManagement.resetPassword', 'Are you sure you want to reset password for {name}?', {
+              name: user?.display_name || ''
+            })}
+          </Dialog.Title>
+        </Dialog.Header>
+
+        {/* TODO: investigate how to make user login bold with font-weight: 600 */}
+        <div className="flex flex-col gap-y-7">
+          <span>
+            {generatePassword
+              ? t(
+                  'views:userManagement.passwordGeneratedMessage',
+                  "Your password has been generated. Please make sure to copy and store your password somewhere safe, you won't be able to see it again."
+                )
+              : t(
+                  'views:userManagement.resetPasswordMessage',
+                  'A new password will be generated to assist {name} in resetting their current password.',
+                  { name: user?.display_name || '' }
+                )}
+          </span>
+
+          {generatePassword && (
+            <Input
```

Comment:
Just to confirm, the user who will be generating a password is unlikely to be the user who the password is for?

Reply:
Yes that's right, the admin can generate or change the password for any other user